### PR TITLE
A crash fix, an optimisation, some other suggestions

### PR DESCRIPTION
**Pull request by jlnr**
I've mostly wanted to fix a crash in `enqueueObjectsFromArray:`, but I've added a bunch of other stuff as well.

### CODE REVIEW

File: LDOCappedQueue/Classes/LDOCappedQueue.m [hunk 1]
```diff
@@ -30,25 +30,26 @@ - (NSUInteger)count
...
-- (NSArray<id> *)allObjects
+- (NSArray *)allObjects
 {
     return [self.buffer copy];
 }
 
 - (void)enqueueObject:(nonnull id)object
 {
-    if (self.count + 1 > self.capacity) {
+    if (self.count == self.capacity) {
```
Comment (jlnr) [inline LDOCappedQueue/Classes/LDOCappedQueue.m:40]: I think this one looks more like `if (queue_is_full)`

Comment (sebastianludwig) [reply LDOCappedQueue/Classes/LDOCappedQueue.m:40]: I initially had that and changed it to be consistent with `enqueueObjectsFromArray:` - but I agree.

File: LDOCappedQueue/Classes/LDOCappedQueue.m [hunk 2]
```diff
@@ -30,25 +30,26 @@ - (NSUInteger)count
...
     [self.buffer addObject:object];
 }
 
-- (void)enqueueObjectsFromArray:(NSArray<id> *)array
+- (void)enqueueObjectsFromArray:(NSArray *)array
 {
     NSUInteger newEntriesCount = MIN(array.count, self.capacity);
     if (self.count + newEntriesCount > self.capacity) {
-        [self.buffer removeObjectsInRange:NSMakeRange(0, newEntriesCount)];
+        NSInteger excessObjects = (self.count + newEntriesCount - self.capacity);
```
Comment (sebastianludwig) [inline LDOCappedQueue/Classes/LDOCappedQueue.m:51]: unnecessary `(` ;-)

Comment (sebastianludwig) [inline LDOCappedQueue/Classes/LDOCappedQueue.m:51]: test: second conversation about the same line

**Review by sebastianludwig — APPROVED** [review]
perfect, thanks! 👍

**Pull request merged**